Sources/MapboxNavigation/NavigationMapView+RoadNameLabeling.swift

```diff
@@ -0,0 +1,171 @@
+import MapboxMaps
+import MapboxDirections
+import MapboxCoreNavigation
+
+extension NavigationMapView {
+    
+    func labelCurrentRoadFeature(at location: CLLocation,
```

Comment:
#2548 tracks consolidating and exposing the logic for labeling the current road name.

Reply:
@S2Ler, `NavigationMapView.labelCurrentRoadFeature(at:router:wayNameView:roadNameFromStatus:)` is already reused by iOS and CarPlay. Can you elaborate regarding: 

> The original code already has a bug that incorrectly hides road labels.

`WayNameView` is hidden whenever road name info is missing and we were not able to query it. It'll be shown back whenever is such info is available once again.

Reply:
> NavigationMapView.labelCurrentRoadFeature(at:router:wayNameView:roadNameFromStatus:) is already reused by iOS and CarPlay.

I'm sorry, missed that part, that's cool that there is no duplication 👍

Reply:
> `WayNameView` is hidden whenever road name info is missing and we were not able to query it. It'll be shown back whenever is such info is available once again.

I haven't got a chance to create a ticket, but on the freeways, I've noticed that road name disappears even though it shouldn't.

Reply:
@MaximAlien  #3594 created a bug for this.

Reply:
Thanks. Since querying depends on the zoom level we'll have to find more reliable ways of getting road name information. Hopefully in scope of https://github.com/mapbox/mapbox-navigation-ios/issues/2548 we'll be able to fix that.